restapi/src/main/java/io/stargate/web/docsapi/service/query/search/resolver/impl/AllFiltersResolver.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in comHpliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.stargate.web.docsapi.service.query.search.resolver.impl;
+
+import io.reactivex.rxjava3.core.Flowable;
+import io.reactivex.rxjava3.core.Maybe;
+import io.reactivex.rxjava3.core.Single;
+import io.stargate.db.query.BoundQuery;
+import io.stargate.db.query.Query;
+import io.stargate.web.docsapi.dao.Paginator;
+import io.stargate.web.docsapi.service.DocsApiConfiguration;
+import io.stargate.web.docsapi.service.ExecutionContext;
+import io.stargate.web.docsapi.service.QueryExecutor;
+import io.stargate.web.docsapi.service.RawDocument;
+import io.stargate.web.docsapi.service.query.search.resolver.DocumentsResolver;
+import io.stargate.web.docsapi.service.query.search.resolver.filter.CandidatesFilter;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.tuple.Pair;
+
+public class AllFiltersResolver implements DocumentsResolver {
+
+  private final Collection<CandidatesFilter> candidatesFilters;
+
+  private final DocumentsResolver candidatesResolver;
+
+  public AllFiltersResolver(
+      Function<ExecutionContext, CandidatesFilter> candidatesFilterSupplier,
+      ExecutionContext context,
+      DocumentsResolver candidatesResolver) {
+    this(Collections.singleton(candidatesFilterSupplier), context, candidatesResolver);
+  }
+
+  public AllFiltersResolver(
+      Collection<Function<ExecutionContext, CandidatesFilter>> candidatesFilterSuppliers,
+      ExecutionContext context,
+      DocumentsResolver candidatesResolver) {
+    ExecutionContext nested = context.nested("PARALLEL [ALL OF]");
```

Comment:
I think this is a valuable optimization for use cases where latency is important.

However, this approach will likely generate more backend load. 

Would it be too hard to add a request parameter (hint) to control whether parallel execution is allow or preferred or not?

We can certainly do it in a separate PR, but I tend to think we ought to make it configurable before merging to `master`. WDYT?

Reply:
Yep ticket should be good here.